internal/pkg/pac-go-server/services/catalog.go

```diff
@@ -59,12 +59,20 @@ func CreateCatalog(c *gin.Context) {
 		c.JSON(http.StatusBadRequest, gin.H{"error": fmt.Sprintf("failed to bind request, Error: %v", err.Error())})
 		return
 	}
+
 	logger.Debug("create catalog request", zap.Any("request", catalog))
 	if err := validateCreateCatalogParams(catalog); len(err) > 0 {
 		logger.Error("error in create catalog validation", zap.Errors("errors", err))
 		c.JSON(http.StatusBadRequest, gin.H{"error": fmt.Sprintf("%v", err)})
 		return
 	}
+
+	// if Expiry is not set, use Default value
+	if catalog.Expiry == 0 {
+		logger.Info("catalog expiry not set, using default", zap.String("catalog name", catalog.Name), zap.Int("default expiry", utils.DefaultExpirationDays))
```

Comment:
How about this? admin can edit the catalog and set value to 0, for which we can't say its unset.
```suggestion
		logger.Info("Catalog expiry is set to 0, which is invalid, using default expiry", zap.String("catalogName", catalog.Name), zap.Int("defaultExpiry", utils.DefaultExpirationDays))
```